### ISSUE TITLE
Expression: vector iterators incompatible when using UnaryUnionOp

### ISSUE THREAD

**Issue by sglvladi** [libgeos/geos#584]
I am trying to make use of ``geos/operation/union/UnaryUnionOp`` to compute the unary union of some geometries and running into a MSVC Debug Assertion Failed error when building/running in Debug mode:

```
Debug Assertion Failed!

Program:
C:\Users\sglvladi\source\repos\testgeosunion\x64\Debug\geosd.dll
File: C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.31.31103\include\vector
Line: 186

Expression: vector iterators incompatible`

...
```

Below I have included a sample script that raises the above error (the code has been adjusted from [tests/unit/operation/geounion/UnaryUnionOpTest.cpp](https://github.com/libgeos/geos/blob/main/tests/unit/operation/geounion/UnaryUnionOpTest.cpp)) :

```c++
#include <iostream>
#include <geos/io/WKTReader.h>
#include <geos/geom/GeometryFactory.h>
#include <geos/operation/union/UnaryUnionOp.h>


using namespace geos;


int main() {
    io::WKTReader wktreader;
    geom::GeometryFactory::Ptr gf = geom::GeometryFactory::create();

    std::string geomStr = "MULTILINESTRING ((0.0000000000000000 0.0000000000000000, 0.0000000000000000 5.0000000000000000), (-1.0000000000000000 1.0000000000000000, -1.0000000000000000 3.0000000000000000, 1.0000000000000000 3.0000000000000000, 1.0000000000000000 1.0000000000000000, -1.0000000000000000 1.0000000000000000))";
    std::vector<geom::Geometry*> geoms = { wktreader.read(geomStr).release() };
    geom::Geometry::Ptr borders = operation::geounion::UnaryUnionOp::Union(geoms, *gf);

    std::cout << borders->toString() << std::endl;
    
    return 0;
}
```

The error is raised when running:
```
geom::Geometry::Ptr borders = operation::geounion::UnaryUnionOp::Union(geoms, *gf);
```

The above error does not occur when building/running in Release mode.

For reference, I have installed geos v3.10.0 via vcpkg.

Any help would be much appreciated.

**Comment by pramsey** [comment]
Can you pull the latest 3.10 branch? We did have a little bug with MSVC Debug/Release mode different behaviours that perhaps is gone now?

**Comment by sglvladi** [comment]
Thanks for the swift response. Building the latest 3.10 branch did the trick 👍 

Would you know when this will be pushed into an official release? Thanks

**Comment by dbaston** [comment]
It should be fixed in 3.10.2, released in January.
